core/math/vector3.h

```diff
@@ -512,6 +514,12 @@ void Vector3::normalize() {
 	}
 }
 
+Vector3 Vector3::normalized_nonzero() const {
+	float lengthsq = x * x + y * y + z * z;
+	float length = Math::sqrt(lengthsq);
```

Comment:
Shouldn't this be real_t instead of float?

Now that I am thinking about it, don't we have to change all the alignment stuff based on whether or not real_t is double?
Since otherwise you can still end up in a situation where stuff is split up across multiple cache lines, making this optimization pointless or even counterproductive (in case the compiler would have automatically defaulted to an alignment that made more sense)?

Reply:
Yes, it should be real_t, as for the alignment for double that's what I need to look at, and why it is still a draft.